### PR TITLE
add a Vagrant file for Linux & Win dev VMs

### PR DESCRIPTION
**Pull request by robbkidd**
Much simpler provision scripts than #3 for Qt/C++.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
+Vagrant.configure("2") do |config|
+  config.ssh.forward_agent = true
+
+  config.vm.define "ubuntu" do |node|
+    node.vm.box = "ubuntu/bionic64"
+
+    # Install xfce and virtualbox additions for GUI.
+    # You must reboot this VM after provisioning to get windowed environment.
+    node.vm.provision "shell", inline: <<~DEV_SETUP
+      curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
```
Comment (robbkidd) [inline Vagrantfile]: Currently these provision scripts pull in Node 8, the LTS release.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
+Vagrant.configure("2") do |config|
+  config.ssh.forward_agent = true
+
+  config.vm.define "ubuntu" do |node|
+    node.vm.box = "ubuntu/bionic64"
```
Comment (tyler-ball) [inline Vagrantfile]: Lets use the bento box?

Comment (cheeseplus) [reply Vagrantfile]: image

Comment (robbkidd) [reply Vagrantfile]: Rassafrappit! I don't know why I keep getting tripped up on bringing in bentos.

Yes, of course.

**Comment by robbkidd** [comment]
Bento'd. And changed to Node 10 instead of Node 8.

The Ubuntu VM installs xfce4 which may be insufficient for our tray needs.

**Pull request merged**